template/src/navigation/ScreenService.js

```diff
@@ -17,15 +17,36 @@ export const screenMatch = screen => {
   }
 }
 
-export const optionsMatch = screen => {
-  switch (screen) {
-    // Screen Options
+export const optionsMatch = ({route}) => {
+  const name = route?.name
+  let headerShown = true
+  let headerBackVisible = true
+  let headerTintColor = colors.black
+  let headerTransparent = false
+  let headerBackTitleVisible = false
+  let gestureEnabled = true
+  let animation = 'default'
+  let title = ''
+  let headerLeft = null
+  switch (name) {
     case RouteKey.HomeScreen:
     case RouteKey.HomeStack:
-      return {
-        headerLeft: null,
-      }
+      headerLeft = null
+      break
     default:
-      return {}
+      break
+  }
+  // Screen Options
```

Comment:
I don't think we need generate screen options any more

Reply:
@honghoangsts I think we still need this, in order to configure for `options` for screens
This PR is just to remove the need to pass `RouteKey.<ScreenName>` inside this option generator function

Reply:
I don't think all screen need to add to this function. There are some specific case need to change the header options.

Reply:
Well this is just to put all the custom options for screens in one place. If those screen that don't need to change option can ignore this. But I think we still need it here for other screen that need changing

Reply:
So maybe I will remove the default return object and let every switch case return their own value. Like this
```
export const optionsMatch = ({route}) => {
  const name = route?.name
  switch (name) {
    case RouteKey.HomeScreen:
    case RouteKey.HomeStack:
      return {
        headerLeft: null,
      }
      break
    default:
      return {}
  }
}
```

Reply:
@honghoangsts updated auto generate screen scripts